libp2p/peerinfo.nim

```diff
@@ -28,6 +33,25 @@ type
     agentVersion*: string
     privateKey*: PrivateKey
     publicKey*: PublicKey
+    signedPeerRecord*: Option[Envelope]
+
+proc createSignedPeerRecord(peerId: PeerID, addrs: seq[MultiAddress], key: PrivateKey): Result[Envelope, CryptoError] =
+  ## Creates a signed peer record for this peer:
+  ## a peer routing record according to https://github.com/libp2p/specs/blob/master/RFC/0003-routing-records.md
+  ## in a signed envelope according to https://github.com/libp2p/specs/blob/master/RFC/0002-signed-envelopes.md
+
+  # First create a peer record from the peer info
+  let peerRecord = PeerRecord.init(peerId,
+                                   getTime().toUnix().uint64, # This currently follows the recommended implementation, using unix epoch as seq no.
+                                   addrs)
+
+  # Wrap peer record in envelope and sign
+  let envelope = ? Envelope.init(key,
```

Comment:
I was thinking about this, maybe we should instead create a

Envelope.init(PrivateKey, PeerRecord)
&
Envelope.decodePeerRecord(seq[byte])
(or maybe getField is better, not sure) 

Inside peer_record.nim, and put the constants in it? (don't even need to export them)
wdyt?

Reply:
I agree. Seems cleaner this way. Have gone ahead and moved all initialisation and decoding of signed peer record envelopes to routing_record.nim in 0d5dfbb.

This might not have been your exact intention, so LMK :)